### DISCUSSION TITLE
What to do if I need Security on a 404 page ?

### DISCUSSION THREAD

**Discussion by Adham-A** [symfony/symfony#57394] [Q&A]
Hello everyone,

I'm using Symfony 5.4 and I need to customise error pages.

When the user is not logged in I want to throw a generic page without the application layout and when he is logged in the application layout with an error message.

My main issue is that the Security component is not available on 404 pages because of listeners' order,

Do you know a method to circumvent this ? 
As my app.user in twig ends up always being null

I've tried to search online but no solutions seemed satisfying with my issue

**Comment by MatTheCat** [comment]
Hello,

you can fetch the error page’s content from an existing controller, using JavaScript or [`render`](https://symfony.com/doc/current/reference/twig_reference.html#render).

**Comment by Trismegiste** [comment]
You can add a priority on listeners

> There is an optional attribute for the kernel.event_listener tag called priority, which is a positive or negative integer that defaults to 0 and it controls the order in which listeners are executed (the higher the number, the earlier a listener is executed). This is useful when you need to guarantee that one listener is executed before another. The priorities of the internal Symfony listeners usually range from -256 to 256 but your own listeners can use any positive or negative integer.

https://symfony.com/doc/5.x/event_dispatcher.html
